source/extensions/filters/http/ext_authz/ext_authz.cc

```diff
@@ -110,6 +110,7 @@ Http::FilterHeadersStatus Filter::decodeHeaders(Http::RequestHeaderMap& headers,
     if (config_->denyAtDisable()) {
       ENVOY_STREAM_LOG(trace, "ext_authz filter is disabled. Deny the request.",
```

Comment:
Looks to me like you almost nailed the requirements for source/docs/upstream_filters.md
the one I think is outstanding is 

Any code accessing the downstream connection checks to make sure it is present.

I see a bunch of decoder_callbacks_->connection() which I _think_ won't work for shadow requests.  
I think we should put in connection checks, add a test of shadowing that would have crashed without those changes, and comment the limitation in the authz filter docs.

Reply:
Fixed - it turns out ext_authz filter calls into a common `createHttpCheck()` that also references a bunch of things on the downstream connection, like the peer certificate and the TLS session, so I've also config-disallowed those fields from being set on the upstream filter chain.